src/miner_lora_light.erl

```diff
@@ -0,0 +1,819 @@
+    chain = undefined :: undefined | blockchain:blockchain(),
+    radio_udp_bind_ip,
+    radio_udp_bind_port,
+    cur_poc_challenger_type = undefined :: undefined | validator,
+    following_chain = true :: boolean
+}).
+
+-type state() :: #state{}.
+-type gateway() :: #gateway{}.
+-type helium_packet() :: #packet_pb{}.
+-type freq_data() :: {Region::atom(), Freqs::[float]}.
+
+-define(COUNTRY_FREQ_DATA, country_freq_data).
+
+%% in meters
+-define(MAX_WANDER_DIST, 200).
+
+%% Maximum `tmst` counter value reported by an SX130x concentrator
+%% IC. This is a raw [1] counter value with the following
+%% characteristics:
+%%
+%% - unsigned
+%% - counts upwards
+%% - 32 bits
+%% - increments at 1 MHz
+%%
+%% [1]: On SX1301 it is a raw value. On SX1302 it is a 32 bit value
+%% counting at 32 MHz, but the SX1302 HAL throws away 5 bits to match
+%% SX1301's behavior.
+%%
+%% Equivalent `(2^32)-1`
+-define(MAX_TMST_VAL, 4294967295).
+
+-ifdef(TEST).
+-define(REG_DOMAIN_TIMEOUT, 1000).
+-else.
+-define(REG_DOMAIN_TIMEOUT, 30000).
+-endif.
+
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(Args) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, Args, []).
+
+%% @doc used to handle state channel responses
+-spec handle_response(blockchain_state_channel_response_v1:response()) -> ok | {error, any()}.
+handle_response(Resp) ->
+    case blockchain_state_channel_response_v1:downlink(Resp) of
+        undefined ->
+            ok;
+        Packet ->
+            send(Packet)
+    end.
+
+-spec send(helium_packet()) -> ok | {error, any()}.
+send(#packet_pb{payload=Payload, frequency=Freq, timestamp=When, signal_strength=Power, datarate=DataRate}=Packet) ->
+    lager:debug("got download packet ~p via freq ~p", [Packet, Freq]),
+    %% this is used for downlink packets that have been assigned a downlink frequency by the router, so just use the supplied frequency
+    ChannelSelectorFun = fun(_FreqList) -> Freq end,
+    gen_server:call(?MODULE, {send, Payload, When, ChannelSelectorFun, DataRate, Power, true, Packet}, 11000).
+
+-spec send_poc(binary(), any(), function(), iolist(), any()) -> ok | {error, any()} | {warning, any()}.
+send_poc(Payload, When, ChannelSelectorFun, DataRate, Power) ->
+    gen_server:call(?MODULE, {send, Payload, When, ChannelSelectorFun, DataRate, Power, false, undefined}, 11000).
+
+-spec port() -> {ok, inet:port_number()} | {error, any()}.
+port() ->
+    gen_server:call(?MODULE, port, 11000).
+
+-spec position() -> {ok, {float(), float()}} |
+                    {ok, bad_assert, {float(), float()}} |
+                    {error, any()}.
+position() ->
+    try
+        gen_server:call(?MODULE, position, infinity)
+    catch _:_ ->
+            {error, no_fix}
+    end.
+
+-spec location_ok() -> true | false.
```

Comment:
this can be removed, we never activated it, so there's no point in carrying it forward into the new world.